nucypher/utilities/sandbox/constants.py

```diff
@@ -15,13 +15,39 @@
 along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
 """
 
+
+import contextlib
 import os
 
+import socket
+
 from nucypher.blockchain.eth.constants import DISPATCHER_SECRET_LENGTH, M
+from nucypher.config.characters import UrsulaConfiguration
 
-MOCK_KNOWN_URSULAS_CACHE = {}
 
-MOCK_URSULA_STARTING_PORT = 49152
+def select_test_port() -> int:
```

Comment:
I know we discussed this, but I'm not sure I see a good reason for this. This actually seems like it would cause indeterminacy for running nodes.

It's easy for me to say, "Oh, I need to work with port 9151" and work around what I currently have running, or modify something else. If this just automatically and uncertainly selects a port, I'm going to be confused when an IDS or something tries to figure out what is running on that port.

Reply:
First off this function certainly does not belong in a module named `constants`.... I'm unsure where to put it.

>  This actually seems like it would cause indeterminacy for running nodes.

This function is only called once, the first time the module is imported, an since it's *only* ever used in test runs, is effectively session-based.

> It's easy for me to say, "Oh, I need to work with port 9151" and work around what I currently have running, or modify something else. 

The users production configuration on the disk, and live running node are sensitive and must not be disturbed by tests in any way.  

> I'm going to be confused when an IDS or something tries to figure out what is running on that port.

Perhaps if you are running an IDS that complains about calls to and from localhost?

Reply:
Ah, this is only for testing. Disregard my comments.